website/source/docs/providers/google/r/compute_instance_template.html.markdown

```diff
@@ -136,6 +136,11 @@ The following arguments are supported:
 * `metadata` - (Optional) Metadata key/value pairs to make available from
     within instances created from this template.
 
+* `metadata_startup_script` - (Optional) An alternative to using the
+    startup-script metadata key, mostly to match the compute_instance resource.
+    This replaces the startup-script metadata key on the created instance and 
+    thus the two mechanisms are not allowed to be used simultaneously.
```

Comment:
Ah yep you are right. So the read method filters out `startup-script` if and only if `metadata_startup_script` is set on both resources.

If the user clears out `metadata_startup_script` in the config there might be an odd edge case there, but I think that's far enough off the beaten path to accept some slightly confusing behavior.

Thanks for helping clarify! 👍